_posts/2023-04-13-denoising-diffusion/denoising-diffusion.Rmd

```diff
@@ -0,0 +1,208 @@
+author:
+  - name: Daniel Falbel
+    affiliation: Posit
+    affiliation_url: https://www.posit.co/
+  - name: Sigrid Keydana
+    affiliation: Posit
+    affiliation_url: https://www.posit.co/
+slug: falbelkeydiffusion
+date: 2023-04-13
+bibliography: references.bib
+categories:
+  - Torch
+  - R
+  - Image Recognition & Image Processing
+  - Generative Models
+output:
+  distill::distill_article:
+    self_contained: false
+    toc: true
+preview: images/flowers.png
+editor_options: 
+  markdown: 
+    wrap: 72
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE, fig.width = 6, fig.height = 6)
+```
+
+#### *A Preamble, sort of*
+
+*As we're writing this -- it's April, 2023 -- it is hard to overstate
+the attention going to, the hopes associated with, and the fears
+surrounding deep-learning-powered image and text generation. Impacts on
+society, politics, and human well-being deserve more than a short,
+dutiful paragraph. We thus defer appropriate treatment of this topic to
+dedicated publications, and would just like to say one thing: The more
+you know, the better; the less you'll be impressed by over-simplifying,
+context-neglecting statements made by public figures; the easier it will
+be for you to take your own stance on the subject. That said, we begin.*
+
+<br />
+
+In this post, we introduce an R `torch` implementation of *Denoising
+Diffusion Implicit Models* (@song2020). The code is on
+[GitHub](https://github.com/dfalbel/denoising-diffusion), and comes with
+an extensive README detailing everything from mathematical underpinnings
+via implementation choices and code organization to model training and
+sample generation. Here, we give a high-level overview, situating the
+algorithm in the broader context of generative deep learning. Please
+feel free to consult the README for any details you're particularly
+interested in!
+
+## Diffusion models in context: Generative deep learning
+
+In generative deep learning, models are trained to generate new
+exemplars that could likely come from some familiar distribution: the
+distribution of landscape images, say, or Polish verse. While diffusion
+is all the hype now, the last decade had much attention go to other
+approaches, or families of approaches. Let's quickly enumerate some of
+the most talked-about, and give a quick characterization.
+
+First, **diffusion models** themselves. Diffusion, the general term,
+designates entities (molecules, for example) spreading from areas of
+higher concentration to lower-concentration ones, thereby increasing
+entropy. Making the leap from entropy to information, information is
+lost. In diffusion models, this information loss is intentional: In a
+"forward" process, a sample is taken and successively transformed into
+(Gaussian, usually) noise. A "reverse" process then is supposed to take
+an instance of noise, and sequentially de-noise it until it looks like
```

Comment:
OK, so when I read the title to this, I wanted a hyphen in de-noise (it's just easier for me to read). From a cursory search, it seems like the hyphen-free version (denoise/denoising) is definitely in use. So, it's your call, but I'd make it consistent.

Reply:
Yeah, I prefer the hyphenated version as well :-))
(Definitely the other is used more often, but still!)